src/video/wayland/SDL_waylandwindow.c

```diff
@@ -1486,6 +1487,41 @@ static const struct wp_fractional_scale_v1_listener fractional_scale_listener =
     handle_preferred_fractional_scale
 };
 
+static void frog_preferred_metadata_handler(void *data, struct frog_color_managed_surface *frog_color_managed_surface, uint32_t transfer_function,
+                                            uint32_t output_display_primary_red_x, uint32_t output_display_primary_red_y,
+                                            uint32_t output_display_primary_green_x, uint32_t output_display_primary_green_y,
+                                            uint32_t output_display_primary_blue_x, uint32_t output_display_primary_blue_y,
+                                            uint32_t output_white_point_x, uint32_t output_white_point_y,
+                                            uint32_t max_luminance, uint32_t min_luminance,
+                                            uint32_t max_full_frame_luminance)
+{
+    SDL_WindowData *wind = (SDL_WindowData *)data;
+    SDL_HDROutputProperties HDR;
+
+    SDL_zero(HDR);
+
+    /* The reference white level seems to be 1.0 */
+    switch (transfer_function) {
+    case FROG_COLOR_MANAGED_SURFACE_TRANSFER_FUNCTION_ST2084_PQ:
+    case FROG_COLOR_MANAGED_SURFACE_TRANSFER_FUNCTION_SCRGB_LINEAR:
```

Comment:
This will be different between ST2084 and SCRGB. There aren't currently any renderers that support PQ output, but the "correct" value for PQ is white level = 100 (or 200, depending on who you ask), clamped to the max luminance, and headroom divided by that.

Reply:
203 nit is pretty much per spec (ITU-R BT.2408). How do people come up with 100 nit?

Reply:
I don't have access to the spec, but the original ST2084 spec supposedly defined diffuse white as 100 nits:
https://lightillusion.com/what_is_hdr.html

Reply:
[The latest ITU-R BT.2408-7 report:](https://www.itu.int/dms_pub/itu-r/opb/rep/R-REP-BT.2408-7-2023-PDF-E.pdf)
>The reference level, HDR Reference White, is defined in this Report as the nominal signal level obtained from an HDR camera and a 100% reflectance white card resulting in a nominal luminance of 203 cd/m2 on a PQ display or on an HLG display that has a nominal peak luminance capability of 1 000 cd/m2.

The publicly available, older Dolby docs that Google turns up have 100 as the reference level.

203 seems to be the current spec, and it looks like the final Wayland color protocol will [communicate the reference levels](https://gitlab.freedesktop.org/swick/wayland-protocols/-/merge_requests/74).

Reply:
> The publicly available, older Dolby docs that Google turns up have 100 as the reference level.

Out of curiosity, do you have a link?

Reply:
"Dolby PQ reference white level" turned up this (and some various discussions that also reference this or their other, older specs).
https://professional.dolby.com/siteassets/pdfs/operational-guidelines-for-pq.pdf

[The Netflix HDR mastering guidelines](https://partnerhelp.netflixstudios.com/hc/en-us/articles/360000599948-Dolby-Vision-HDR-Mastering-Guidelines) cite an SDR reference of 100 nits.

There seems to be quite a bit of conflicting information out there.